### DISCUSSION TITLE
capacity increasing over time for LFP battery

### DISCUSSION THREAD

**Discussion by skywo1f** [pybamm-team/PyBaMM#2472] [General]
I ran the updated Prada2013 parameters over many cycles:
`
import pybamm

#chemistry = pybamm.parameter_sets.Prada2013
param = pybamm.ParameterValues("Prada2013")

pybamm.set_logging_level("NOTICE")

exp = pybamm.Experiment(
    [
        (
            "Charge at 1 C until 3.65 V",
            "Hold at 3.65 V until C/10",
            "Rest for 5 minutes",
            "Discharge at 1 C until 2.5 V",
            "Rest for 5 minutes",
        )
    ] * 50,
    termination="80% capacity"
    )
model = pybamm.lithium_ion.SPM()

sim = pybamm.Simulation(model,experiment=exp,parameter_values=param)
sol = sim.solve([0, 3600],initial_soc=0)  # solve for 1 hour
print(sim.parameter_values)

pybamm.plot_summary_variables(sol)
`
But I found that the capacity increased over the course of the cycles...
![lfpCapacity](https://user-images.githubusercontent.com/4000822/202745954-e9bca1d9-ddce-4637-b384-1b3041835a41.png)

**Comment by dion-w** [comment]
Seems like a numerical issue to me. If you're looking at the capacity plot, above it says "1e-8+1.2242440000". Which means, the numbers on the left are to be multiplied with 1e-8 and added with 1.22. So overall the capacity is increasing with an mount of 0.1 *1e-9 over 50 cycles, which would be 0.1 _**nano**_ Ah. So i think the simulation works just fine. The very detailed default plotting just shows you the very slight increase of capacity probably due to computationally completely normal rounding errors in the code.

**Reply by skywo1f** [reply]
Shouldn't the capacity be strictly decreasing over time? Is there a parameter that Prada2013 is missing (or has wrong) which would allow me to simulate capacity loss over cycles?

**Comment by valentinsulzer** [comment]
@dion-w 's answer might be true in this case. However I think there is also a more general problem. In some of the parameter sets, the total amount of lithium (n_Li) is too large (larger than the theoretical maximum for either electrode), and a consequence of this is that in the capacity calculation algorithm decreasing n_Li increases theoretical capacity. 

Below is a case where `n_Li` is less than the value that gives maximum capacity, but you can see that if it were larger than that (around 0.010) then decreasing it would increase theoretical capacity.

So it's a bug in the parameters

![image](https://user-images.githubusercontent.com/20817509/202753320-e368d61b-4c2f-4b45-8307-3b403c58fec4.png)

**Reply by skywo1f** [reply]
I did notice that the default concentrations are very different for positive vs negative electrodes:
`
        "Initial concentration in negative electrode [mol.m-3]": 28831.45783,
        "Initial concentration in positive electrode [mol.m-3]": 35.3766672,
`

 Either way, is this one of the values that is too high and not letting the capacity drop over cycles?
 
 ...unless that is just another way of saying that the battery starts out charged.

**Reply by manjunathnilugal** [reply]
Sir, is there any methodology to get over this issue, so that degradation occurs from the starting cycle.

**Comment by manjunathnilugal** [comment]
Dear all,
Regarding above mentioned issue, The Loss of capacity is increasing, but capacity is not decreasing.
```
import pybamm
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

model = pybamm.lithium_ion.DFN({"particle":"Fickian diffusion",
                                "SEI" : "reaction limited",
                                "SEI film resistance": "distributed",
                                "SEI porosity change": "false",
                                "lithium plating":"none"})

param = pybamm.ParameterValues("Ramadass2004")

param.update({
    "Inner SEI reaction proportion": 0.5,
    "SEI growth activation energy [J.mol-1]" : 0,
    "SEI reaction exchange current density [A.m-2]": 1.5e-6,
    "SEI resistivity [Ohm.m]": 200000,
    "Initial inner SEI thickness [m]": 2.5e-9,
    "Initial outer SEI thickness [m]": 2.5e-9,
    "Inner SEI partial molar volume [m3.mol-1]": 9.585e-5,
    "Outer SEI partial molar volume [m3.mol-1]": 9.585e-5,
    "Ratio of lithium moles to SEI moles": 2,
    "Lower voltage cut-off [V]" : 1.5,
}, check_already_exists=False)

experiment = pybamm.Experiment([
    (f"Discharge at 1C until 2V",
     f"Charge at 1C until 4.2V",
     f"Hold at 4.2V until 0.1C"
    )
]*500,
termination="80% capacity"
)

var_pts = {
    "x_n": 30,  # negative electrode
    "x_s": 30,  # separator
    "x_p": 30,  # positive electrode
    "r_n": 30,  # negative particle
    "r_p": 30,  # positive particle
}

safe_solver = pybamm.CasadiSolver(mode = "safe", rtol=1e-3, atol=1e-3, dt_max=0.5)
sim = pybamm.Simulation(model,parameter_values = param, solver = safe_solver, experiment = experiment, var_pts=var_pts)
solution = sim.solve()
pybamm.plot_summary_variables(solution)
```
![image](https://user-images.githubusercontent.com/82285213/202762001-6d942828-3c84-426b-830a-5c42e07086ed.png)

Even though the loss of capacity is very less, it should reflect in the capacity loss, or I am missing something very important in including in degradation model.
I am facing the similar issue with my own set of parameters for all sei models, i.e reaction limited, ec reaction limited, electron tunnelling, etc

**Reply by skywo1f** [reply]
It's weird because the "simulating-long-experiments" notebook is able to simulate capacity loss over cycles

**Reply by skywo1f** [reply]
it looks like it is the 
`model  = pybamm.lithium_ion.SPM({"SEI": "ec reaction limited"})
`
parameter that decides if batteries will degrade over time. Though this doesn't work for Prada2013

**Comment by manjunathnilugal** [comment]
@tinosulzer is there any work around to get the capacity degradation of a battery vs cycle no. in PyBaMm framework.

**Reply by valentinsulzer** [reply]
Try `solution.summary_variables["Measured capacity [A.h]"]`
